rubyparser.h

```diff
@@ -54,6 +54,15 @@ typedef struct rb_parser_string {
     char *ptr;
 } rb_parser_string_t;
 
+/*
+ * Array-like object for parser
+ */
+typedef struct rb_parser_ary {
+    uintptr_t *data;
```

Comment:
Why this `data` is a pointer to `uintptr_t` instead of  `rb_parser_ast_token_t`?
Also `val` of  `rb_parser_ary_push`.

Reply:
We plan to have rb_parser_ary store elements of arbitrary types. However, after talking to Kaneko-san, we decided to limit it to the `rb_parser_ast_token_t*` type for now.
https://github.com/ruby/ruby/pull/10004/commits/85ff09cb381da4be277823aba41de8596c0ecc1c
Thank you for the review!